bin/v-import-directadmin

```diff
@@ -63,42 +64,79 @@ tput sgr0
 tput setaf 2
 echo "Checking provided file..."
 tput sgr0
-if file $backup_file | grep -q -c "gzip compressed data,"; then
-	tput setaf 2
-	echo "OK - Gziped File"
-	tput sgr0
-	if [ ! -d /backup/${tmp_dir} ]; then
-		echo "Creating tmp.."
-		mkdir /backup/${tmp_dir}
-	fi
-	echo "Extracting backup..."
-	if [ "$debug" != 0 ]; then
-		tar xzvf $backup_file -C /backup/${tmp_dir} 2>&1 \
-			| while read extracted_file; do
-				ex=$((ex + 1))
-				echo -en "wait... $ex files extracted\r"
-			done
-	else
-		tar xzf $backup_file -C /backup/${tmp_dir}
-	fi
-	if [ $? -eq 0 ]; then
-		tput setaf 2
-		echo "Backup extracted whitout errors..."
-		tput sgr0
-	else
-		tput setaf 1
-		echo "Error on backup extraction, check your file, try extract it manually"
-		tput sgr0
-		delete_tmp
-		exit 1
-	fi
+
+if file "$backup_file" | grep -q -c "gzip compressed data"; then
+    tput setaf 2
+    echo "OK - Gzipped File"
+    tput sgr0
+
+    if [ ! -d /backup/${tmp_dir} ]; then
+        echo "Creating temporary directory..."
+        mkdir /backup/${tmp_dir}
+    fi
+
+    echo "Extracting backup..."
+    if [ "$debug" != 0 ]; then
+        tar xzvf "$backup_file" -C /backup/${tmp_dir} 2>&1 | while read -r extracted_file; do
+            ex=$((ex + 1))
+            echo -en "wait... $ex files extracted\r"
+        done
+    else
+        tar xzf "$backup_file" -C /backup/${tmp_dir}
+    fi
+
+    if [ $? -eq 0 ]; then
+        tput setaf 2
+        echo "Backup extracted without errors..."
+        tput sgr0
+    else
+        tput setaf 1
+        echo "Error on backup extraction, check your file and try extracting it manually"
+        tput sgr0
+        delete_tmp
+        exit 1
+    fi
+
+elif file "$backup_file" | grep -q -c "Zstandard compressed data"; then
+    tput setaf 2
+    echo "OK - Zstandard Compressed File"
+    tput sgr0
+
+    if [ ! -d /backup/${tmp_dir} ]; then
+        echo "Creating temporary directory..."
+        mkdir /backup/${tmp_dir}
+    fi
+
+    echo "Extracting backup..."
+    if [ "$debug" != 0 ]; then
+        tar --use-compress-program=unzstd -xvf "$backup_file" -C /backup/${tmp_dir} 2>&1 | while read -r extracted_file; do
```

Comment:
Any reason to not use  pzstd ?

Reply:
Thanks for the reply, have changed the code to pzstd, there was no particular reason to use unzstd. Also have revert back the code for the binaries check